content/id/docs/concepts/containers/container-lifecycle-hooks.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Lifecyle Hook pada Kontainer
+content_template: templates/concept
+weight: 30
+---
+
+{{% capture overview %}}
+
+Laman ini menjelaskan bagaimana semua Kontainer yang diatur kubelet menggunakan *framework lifecycle hook*
+untuk menjalankan kode yang di-*trigger* oleh *event* selama *lifecycle* berlangsung.
+
+{{% /capture %}}
+
+
+{{% capture body %}}
+
+## Ikhtisar
+
+Kubernetes menyediakan *hook* untuk *lifecycle* Kontainer. Hal ini sejalan dengan *framework* bahasa 
+pemrograman kebanyakan yang memiliki *hook* untuk *lifecycle* komponen, seperti Angular contohnya.
+*Hook* tersebut digunakan Kontainer untuk selalu siap menerima *event* selama *lifecycle* dan
+menjalankan kode yang diimplementasi pada suatu *handler*, ketika *hook lifecycle* terkait telah dieksekusi.
+
+## Jenis-jenis *hook* pada Kontainer
+
+Ada dua jenis *hook* yang diekspos pada Kontainer:
+
+`PostStart`
+
+*Hook* ini dijalankan segera setelah suatu kontainer dibuat.
+Hanya saja, tidak ada jaminan bahwa *hook* akan tereksekusi sebelum `ENTRYPOINT` dari kontainer.
+Tidak ada parameter yang diberikan pada *handler*.
+
+`PreStop`
+
+*Hook* ini akan dipanggil sesaat sebelum kontainer dimatikan, karena suatu *request* API atau *event* pengaturan,
+contohnya kegagalan pada *liveness probe*, *preemption*, perebutan *resource*, dan lainnya.
+Sebuah panggilan untuk *hook* `PreStop` akan gagal jika kontainer tersebut telah ada pada *state terminate* atau *complete*.
+Hal ini bersifat *blocking*, yang artinya *synchronous*, harus menunggu eksekusi selesai, sebelum melakukan panggilan
```

Comment:
wondering if we kan use `sinkronis` to replace `synchronous` wdyt @qbl ?

Reply:
we agreed sinkron dan tidak sinkron, I'm changing this